staticaddr/deposit/actions.go

```diff
@@ -0,0 +1,140 @@
+package deposit
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop/fsm"
+	"github.com/lightninglabs/loop/staticaddr/script"
+)
+
+const (
+	defaultConfTarget = 3
+)
+
+// PublishDepositExpirySweepAction creates and publishes the timeout transaction
+// that spends the deposit from the static address timeout leaf to the
+// predefined timeout sweep pkscript.
+func (f *FSM) PublishDepositExpirySweepAction(_ fsm.EventContext) fsm.EventType {
+	msgTx := wire.NewMsgTx(2)
+
+	params, err := f.cfg.AddressManager.GetStaticAddressParameters(f.ctx)
+	if err != nil {
+		return fsm.OnError
+	}
+
+	// Add the deposit outpoint as input to the transaction.
+	msgTx.AddTxIn(&wire.TxIn{
+		PreviousOutPoint: f.deposit.OutPoint,
+		Sequence:         params.Expiry,
+		SignatureScript:  nil,
+	})
+
+	// Estimate the fee rate of an expiry spend transaction.
+	feeRateEstimator, err := f.cfg.WalletKit.EstimateFeeRate(
+		f.ctx, defaultConfTarget,
+	)
+	if err != nil {
+		return f.HandleError(fmt.Errorf("timeout sweep fee "+
+			"estimation failed: %v", err))
+	}
+
+	weight := script.ExpirySpendWeight()
+
+	fee := feeRateEstimator.FeeForWeight(weight)
+
+	// We cap the fee at 20% of the deposit value.
+	if fee > f.deposit.Value/5 {
```

Comment:
This would be a conf flag that could be used in a bunch of other places, for example in reservations or loop-in expiries. I will open an issue to address this in a separate PR.

Reply:
Added an issue to track this https://github.com/lightninglabs/loop/issues/729